eduNEXT/xblock-mindmap
# feat: use fields to store xblock state

## Description

This PR updates the implementation of the xblock to use xblock fields to store the state of the static mindmap and the student mindmap.

This PR fixes a bug in which if we add two instances of the xblock to the same subsection it would crash.

This PR does a major refactor simplifying code to get the context and to render the views

## Test instructions:

1. In the CMS
2. Create a static mindmap.
3. You should be able to edit it on the Edit page, which should be saved once you click Save.
4. Create an editable mindmap.
5. You should be able to edit it on the Edit page; students can only see it in the LMS.
6. When learners load the XBlock, they will get the initial Xblock configured in Studio and can start working on it and saving it.
7. Students should be able to see the instructions to use the xblock.
8. Students can reload the page and see their updated version of the xblock


# TODO

- Update docstrings
- Add tests


**Merge checklist:**
Check off if complete *or* not applicable:
